src/ukify/ukify.py

```diff
@@ -729,6 +744,108 @@ def make_uki(opts):
     print(f"Wrote {'signed' if sign_args_present else 'unsigned'} {opts.output}")
 
 
+ONE_DAY = datetime.timedelta(1, 0, 0)
+
+
+@contextlib.contextmanager
+def temporary_umask(mask: int):
+    # Drop <mask> bits from umask
+    old = os.umask(0)
+    os.umask(old | mask)
+    try:
+        yield
+    finally:
+        os.umask(old)
+
+
+def generate_key_cert_pair(
+        common_name: str,
+        keylength: int = 2048,
+        valid_days: int = 365 * 10,  # TODO: can we drop the expiration date?
+) -> Tuple[bytes]:
+
+    from cryptography import x509
+    import cryptography.hazmat.primitives as hp
+
+    key = hp.asymmetric.rsa.generate_private_key(
```

Comment:
there should be a comment here that explains the choice of RSA here and the key size, i.e. the fact that UEFI SecureBoot in the wild only really supports RSA up to some key length (2k?)

Reply:
I just copied what our docs and/or mkosi was doing. I'd be happy to add something, but I don't really know too enough to write the text myself.

Reply:
msft says rsa 2k:

https://learn.microsoft.com/en-us/windows-hardware/manufacture/desktop/windows-secure-boot-key-creation-and-management-guidance?view=windows-11#12-public-key-cryptography

maybe just add a link to that as a comment

Reply:
Added now.